components/omega/src/ocn/VertMix.cpp

```diff
@@ -0,0 +1,270 @@
+//===-- ocn/VertMix.cpp - Vertical Mixing Coefficients -----------*- C++
+//-*-===//
+//
+// The VertMix class is responsible for managing the calculation of the
+// vertical diffusivity and viscosity needed for the vertical mixing.
+// It currently has background, convective, and shear mixing options, and
+// they can be additively combined, depending on configuration options. It
+// contains arrays that store the vertical top-of-cell diffusivity and
+// viscosity values for each cell and vertical level.
+//
+//===----------------------------------------------------------------------===//
+
+#include "VertMix.h"
+#include "DataTypes.h"
+#include "HorzMesh.h"
+
+namespace OMEGA {
+
+ShearMix::ShearMix(const HorzMesh *Mesh, const VertCoord *VCoord)
+    : NVertLayers(VCoord->NVertLayers), ZMid(VCoord->ZMid),
+      NEdgesOnCell(Mesh->NEdgesOnCell), EdgesOnCell(Mesh->EdgesOnCell),
+      DvEdge(Mesh->DvEdge), DcEdge(Mesh->DcEdge), AreaCell(Mesh->AreaCell) {}
+
+ConvectiveMix::ConvectiveMix(const VertCoord *VCoord)
+    : NVertLayers(VCoord->NVertLayers) {}
+
+/// Constructor for VertMix
+VertMix::VertMix(const std::string &Name_, ///< [in] Name for VertMix object
+                 const HorzMesh *Mesh,     ///< [in] Horizontal mesh
+                 const VertCoord *VCoord   ///< [in] Vertical coordinate
+                 )
+    : ComputeVertMixConv(VCoord), ComputeVertMixShear(Mesh, VCoord) {
+   VertDiff = Array2DReal("VertDiff", Mesh->NCellsAll, VCoord->NVertLayers + 1);
+   VertVisc = Array2DReal("VertVisc", Mesh->NCellsAll, VCoord->NVertLayers + 1);
+   NCellsAll = Mesh->NCellsAll;
+   NChunks   = VCoord->NVertLayers / VecLength;
+   Name      = Name_;
+
+   defineFields();
+}
+
+/// Destructor for VertMix
+VertMix::~VertMix() {}
+
+/// Instance management
+VertMix *VertMix::Instance = nullptr;
+
+/// Get instance of VertMix
+VertMix *VertMix::getInstance() { return Instance; }
+
+/// Destroy instance of VertMix
+void VertMix::destroyInstance() {
+   delete Instance;
+   Instance = nullptr;
+}
+
+/// Initializes the VertMix (Vertical Mixing Coefficients) class and its
+/// options. it ASSUMES that HorzMesh was initialized and initializes the
+/// VertMix class by using the default mesh, reading the config file, and
+/// setting parameters for the background, convective, and/or shear mixing
+/// routines. Returns 0 on success, or an error code if any required option is
+/// missing.
+void VertMix::init() {
+
+   if (!Instance) {
+      Instance = new VertMix("Default", HorzMesh::getDefault(),
+                             VertCoord::getDefault());
+   }
+
+   Error Err; // error code
+
+   /// Retrieve default VertMix
+   VertMix *vertmix = VertMix::getInstance();
```

Comment:
```suggestion
   VertMix *DefVertMix = VertMix::getInstance();
```
Variable names in Omega are capitalized.